core/src/main/java/com/github/jeanbaptistewatenberg/junit5kubernetes/core/Service.java

```diff
@@ -0,0 +1,166 @@
+        }
+    }
+
+    @Override
+    public void create() {
+        try{
+            createdService.set(coreV1Api.createNamespacedService(NAMESPACE, serviceToCreate, null, null, null));
+            final String serviceName = createdService.get().getMetadata().getName();
+
+            if (this.waitStrategy != null) {
+                try (Watch<V1Service> watch = Watch.createWatch(
+                        coreV1Api.getApiClient(),
+                        coreV1Api.listNamespacedServiceCall(NAMESPACE, null, null, null,
+                                null, null, null, null, null, true, null),
+                        new TypeToken<Watch.Response<V1Service>>() {
+                        }.getType())) {
+
+                    this.waitStrategy.apply(watch, createdService.get());
+                } catch (IOException | ApiException e) {
+                    if (e instanceof ApiException) {
+                        throw logAndThrowApiException((ApiException) e);
+                    }
+                    throw new RuntimeException(e);
+                }
+            }
+
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> removeService(serviceName, coreV1Api)));
+            onKubernetesObjectReady();
+        } catch (ApiException e) {
+            throw logAndThrowApiException(e);
+        }
+    }
+
+    @Override
+    public void remove() {
+        V1Service v1Service = this.createdService.get();
+        if (v1Service != null) {
+            String serviceName = v1Service.getMetadata().getName();
+            removeService(serviceName, coreV1Api);
+        }
+    }
+
+    private static void removeService(String serviceName, CoreV1Api coreV1Api) {
+        try {
+            coreV1Api.deleteNamespacedService(serviceName, NAMESPACE, null, null, null, null, null, null);
+        } catch (ApiException e) {
+            throw logAndThrowApiException(e);
+        } catch (JsonSyntaxException e) {
+            if (e.getCause() instanceof IllegalStateException) {
+                IllegalStateException ise = (IllegalStateException) e.getCause();
+                if (ise.getMessage() != null && ise.getMessage().contains("Expected a string but was BEGIN_OBJECT")) {
+                    // Catching exception because of issue https://github.com/kubernetes-client/java/issues/86
+                } else throw e;
+            } else throw e;
+        }
+    }
+
+    @Override
+    public Service withWaitStrategy(WaitStrategy waitStrategy) {
+        this.waitStrategy = waitStrategy;
+        return this;
+    }
+
+    protected static RuntimeException logAndThrowApiException(ApiException e) {
+        LOGGER.severe("Kubernetes API replied with " + e.getCode() + " status code and body " + e.getResponseBody());
+        System.out.println("Kubernetes API replied with " + e.getCode() + " status code and body " + e.getResponseBody());
+        return new RuntimeException(e.getResponseBody(), e);
+    }
```

Comment:
I believe we should be factorized this as well in the abstract class

Reply:
Hi @JeanBaptisteWATENBERG,

thank you for your quick response! Hopefully this week I will find some time to look closer at your Points.

I needed to create a more specific service for the pods. Instead of extending the service config over a pod, I thougth making an explictit service would be clearer.

Best regards
Lars